### PR TITLE
[TH2-5179] migrated to gradle 8.7 and added toml to share dependency versions between projects

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

File: core/build.gradle.kts [hunk 1]
```diff
@@ -1,12 +1,9 @@
 plugins {
-    kotlin("jvm")
-    id("java-test-fixtures")
+    alias(libs.plugins.kotlin)
     id("java-library")
     id("maven-publish")
```
Comment (OptimumCode) [inline core/build.gradle.kts:4]:
```suggestion
    `java-library`
    `maven-publish`
```

File: core/build.gradle.kts [hunk 2]
```diff
@@ -19,50 +16,46 @@ configurations.all {
...
     implementation(project(":grpc-read-db"))
 
-    implementation("com.exactpro.th2:common:5.10.0-dev")
-    implementation("com.exactpro.th2:common-utils:2.2.2-dev")
-    implementation("com.exactpro.th2:lw-data-provider-utils:0.0.1-dev")
+    implementation(libs.th2.common)
+    implementation(libs.th2.common.utils)
+    implementation(libs.th2.lw.data.provider.utils)
 
     implementation("org.slf4j:slf4j-api")
```
Comment (OptimumCode) [inline core/build.gradle.kts:23]: note: Why didn't you move this dependency to libs catalog? Is it special somehow?

**Review by OptimumCode — APPROVED** [review]

File: core/build.gradle.kts (hunk 2 above)
Comment (Nikita-Smirnov-Exactpro) [reply core/build.gradle.kts:23]: I moved all dependency with explicit version to catalog to be able to look at them in one place and change in one place

File: build.gradle.kts [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+    alias(libs.plugins.th2.publish)
 
-    kotlin("jvm") version "1.8.22" apply false
-    kotlin("kapt") version "1.8.22" apply false
-    id("com.exactpro.th2.gradle.grpc") version "0.0.4" apply false
-    id("com.exactpro.th2.gradle.component") version "0.0.4" apply false
+    alias(libs.plugins.kotlin) apply false
+    alias(libs.plugins.kapt) apply false
+    alias(libs.plugins.th2.grpc) apply false
+    alias(libs.plugins.th2.component) apply false
```
Comment (OptimumCode) [inline build.gradle.kts:8]: note: When we add plugins here with `apply false` this is done to specify the version only in one place. If we use dependency catalog there is no point in doing that - we are using the same version in all places automatically

File: app/gradle.properties [hunk 4]
```diff
@@ -1,3 +1,3 @@
 kotlin.code.style=official
-release_version=0.9.0
+release_version=0.9.1
```
Comment (OptimumCode) [inline app/gradle.properties:2]: note: We didn't add any new functionality - why do we increment the version?

**Review by OptimumCode — APPROVED** [review]

File: app/gradle.properties (hunk 4 above)
Comment (Nikita-Smirnov-Exactpro) [reply app/gradle.properties:2]: We solved the `pull continuation failure when column with oracle DATE type is used for update query.` problem

**Comment by Nikita-Smirnov-Exactpro** [comment]
@OptimumCode could you please review again, I have fixed flicker test problem

File: app/build.gradle.kts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 plugins {
     id("application")
-    alias(libs.plugins.th2.component)
+    id("com.exactpro.th2.gradle.component")
```
Comment (OptimumCode) [inline app/build.gradle.kts:3]: note: Why have you decided to go back with declaring id in the gradle file instead of version catalog?

**Review by OptimumCode — APPROVED** [review]

File: app/build.gradle.kts (hunk 5 above)
Comment (Nikita-Smirnov-Exactpro) [reply app/build.gradle.kts]: Reverted. I have unintended incorrectly your comment 
> note: When we add plugins here with apply false this is done to specify the version only in one place. If we use dependency catalog there is no point in doing that - we are using the same version in all places automatically

**Pull request merged**